src/Exception/Binary/Loader/ChainAttemptNotLoadableException.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the `liip/LiipImagineBundle` project.
+ *
+ * (c) https://github.com/liip/LiipImagineBundle/graphs/contributors
+ *
+ * For the full copyright and license information, please view the LICENSE.md
+ * file that was distributed with this source code.
+ */
+
+namespace Liip\ImagineBundle\Exception\Binary\Loader;
+
+use Liip\ImagineBundle\Binary\Loader\LoaderInterface;
+
+final class ChainAttemptNotLoadableException extends NotLoadableException
+{
+    private string          $loaderIndex;
+    private LoaderInterface $loaderClass;
+
+    public function __construct(string $loaderIndex, LoaderInterface $loaderClass, NotLoadableException $loaderException)
+    {
+        $this->loaderIndex = $loaderIndex;
+        $this->loaderClass = $loaderClass;
+
+        parent::__construct($this->compileMessageTxt(), 0, $loaderException);
+    }
+
+    public function getLoaderIndex(): string
+    {
+        return $this->loaderIndex;
+    }
+
+    public function getLoaderClass(): LoaderInterface
+    {
+        return $this->loaderClass;
+    }
+
+    public function getLoaderClassName(): string
```

Comment:
but this one should stay as is, it gives you the classname of the loader, so i like the name.

if we move reflection to the constructor, this will become a simple getter for `$this->loaderClassName`